configstore/index.d.ts

```diff
@@ -1,37 +1,65 @@
-// Type definitions for configstore 0.3.0
+// Type definitions for Configstore 2.1.0
 // Project: https://github.com/yeoman/configstore
-// Definitions by: Bart van der Schoor <https://github.com/Bartvds>
-// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-
-
-/*
- Set an item
- */
-export declare function set(key: string, val: any): void;
-
-/*
- Get an item
- */
-export declare function get(key: string): any;
-
-/*
- Delete an item
- */
-export declare function del(key: string): void;
-
-/*
- Get all items as an object or replace the current config with an object:
-
- conf.all = {
-    hello: 'world'
- };
- */
-export declare var all: Object;
-/*
- Get the item count
- */
-export declare var size: number;
-/*
- Get the path to the config file. Can be used to show the user where the config file is located or even better open it for them.
- */
-export declare var path: string;
+// Definitions by: ArcticLight <https://github.com/ArcticLight/>
+
+export = Configstore
+
+declare class Configstore {
+    constructor(packageName: string, defaults?: any, options?: Configstore.ConfigstoreOptions);
+
+    /**
+     * Get the path to the config file. Can be used to show the user
+     * where it is, or better, open it for them.
+     */
+    path: string;
+
+    /**
+     * Get all items as an object or replace the current config with an object.
+     */
+    all: any;
+
+    /**
+     * Get an item
+     * @param {string} key The string key to get
+     * @return The contents of the config from key $key
+     */
+    get(key: string): any;
+
+    /**
+     * Set an item
+     * @param {string} key The string key
+     * @param {*} val The value to set
+     */
+    set(key: string, val: any): void;
+
+    /**
+     * Set all key/value pairs declared in $values
+     * @param {*} values The values object.
+     */
+    set(values: any): void;
+
+    /**
+     * Determines if a key is present in the config
+     * @param {string} key The string key to test for
+     * @return True if the key is present
+     */
+    has(key: string): boolean;
+
+    /**
+     * Delete an item.
+     * @param {string} key The key to delete
+     */
+    delete(key: string): void;
+
+    /**
+     * Clear the config.
+     * Equivalent to <code>Configstore.all = {};</code>
+     */
+    clear(): void;
+  }
+
+declare namespace Configstore {
+    export interface ConfigstoreOptions {
+        globalConfigPath?: string;
```

Comment:
[docs](https://github.com/yeoman/configstore#globalconfigpath) say it's a boolean.